### PR TITLE
Revert autoinstall

### PR DESCRIPTION
**Pull request by syedriko**

### CODE REVIEW

**Comment by matzew** [comment]
@syedriko   once merged,  please run the `./hack/release.sh` script

thanks

also, I think we need to update the famous PR 420 on the operator-hub

**Comment by syedriko** [comment]
Sure, will do.

File: README.md [hunk 1]
```diff
@@ -1,35 +1,23 @@
...
+    kubectl apply -f deploy/crds/eventing_v1alpha1_knativeeventingkafka_crd.yaml
+    kubectl apply -f deploy/
+    kubectl apply -f deploy/crds/eventing_v1alpha1_knativeeventingkafka_cr.yaml
 
 ## Prerequisites
 
 ### Apache Kafka
 
-This operator installs components that access Apache Kafka, therefore you MUST have a running
+This Operator install components that access Apache Kafka, therefore you MUST to have a running
```
Comment (jcrossley3) [inline README.md:14]: I'm not sure this change improves the grammar. :)

File: README.md [hunk 2]
```diff
@@ -166,23 +132,42 @@ to fire it up on <http://localhost:9000>.
...
+  name: knative-eventing
+  namespace: knative-eventing
+---
 apiVersion: operators.coreos.com/v1alpha1
 kind: Subscription
 metadata:
   name: knative-kafka-operator-sub
   generateName: knative-kafka-operator-
-  namespace: $OPERATOR_NS
+  namespace: knative-eventing
```
Comment (jcrossley3) [inline README.md:157]: Why not keep the sub in `$OPERATOR_NS`? That way, you can avoid having to create the above `Namespace` and `OperatorGroup`.

File: README.md [hunk 3]
```diff
@@ -166,23 +132,42 @@ to fire it up on <http://localhost:9000>.
...
-  sourceNamespace: $OLM_NS
+  sourceNamespace: $OLM
   name: knative-kafka-operator
   channel: alpha
+---
+apiVersion: eventing.knative.dev/v1alpha1
+kind: KnativeEventingKafka
+metadata:
+  name: knative-eventing-kafka
+  namespace: knative-eventing
```
Comment (jcrossley3) [inline README.md:168]: I might omit this and let it reside in the `default` ns.

File: deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -156,9 +156,9 @@ spec:
         serviceAccountName: knative-kafka-operator
     strategy: deployment
   installModes:
-  - supported: false
+  - supported: true
     type: OwnNamespace
-  - supported: false
+  - supported: true
     type: SingleNamespace
```
Comment (jcrossley3) [inline deploy/olm-catalog/knative-kafka-operator/0.6.0/knative-kafka-operator.v0.6.0.clusterserviceversion.yaml:162]: I'm not sure this is better, and it's really orthogonal to the issue of auto-creation.

File: deploy/operator.yaml [hunk 5]
```diff
@@ -22,7 +22,9 @@ spec:
           imagePullPolicy: Always
           env:
             - name: WATCH_NAMESPACE
-              value: ""
```
Comment (jcrossley3) [inline deploy/operator.yaml:25]: Ditto

File: pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go [hunk 6]
```diff
@@ -40,12 +34,16 @@ var (
...
+		return err
+	}
+	return add(mgr, newReconciler(mgr, manifest))
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileKnativeEventingKafka{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+func newReconciler(mgr manager.Manager, man mf.Manifest) reconcile.Reconciler {
+	return &ReconcileKnativeEventingKafka{client: mgr.GetClient(), scheme: mgr.GetScheme(), config: man}
```
Comment (jcrossley3) [inline pkg/controller/knativeeventingkafka/knativeeventingkafka_controller.go:46]: I prefer the `InjectClient` than restoring these. It keeps things more like the osdk scaffolding, which makes updates easier.

**Review by jcrossley3 — CHANGES_REQUESTED** [review]
Removing auto-install is one thing, but I think we should keep the `InjectClient` and AllNamespaces changes in tact.

**Pull request merged**